### PR TITLE
Improve content store readability

### PR DESCRIPTION
**Pull request by random-geek**
| Before | After |
| - | - |
| cstore_ni | cstore_i |
| cstore_ni_dlg | cstore_i_dlg |

My attempt at improving the formatting of the content store. This should be fairly self-explanatory.

### CODE REVIEW

**Comment by ClobberXD** [comment]
Coloured package titles and author names add a lot of clarity. Nice!

**Comment by pauloue** [comment]
The colors are nice, but I think it looks better without the indentation.

**Comment by paramat** [comment]
@rubenwardy

**Comment by rubenwardy** [comment]
I like the colours, but I don't like the indentation

**Comment by paramat** [comment]
The author colouring is possibly slightly too dark a grey.

**Comment by paramat** [comment]
@random-geek see https://github.com/minetest/minetest/issues/7385#issuecomment-431692737

**Comment by paramat** [comment]
I also think it's best without indentation because more space for text is then created.

**Comment by random-geek** [comment]
I will work on applying those changes, although I have been somewhat busy lately.

**Comment by random-geek** [comment]
Changed a few things, as well as the package view dialog.

cstore_1
cstore_2

**Comment by paramat** [comment]
Thanks looks good. Please could you also remove the indentation of the descriptions?

**Comment by ClobberXD** [comment]
As @paramat noted, the bottom of the description text still does get a little too close to the title of the next package. But apart from that, both the content store and the package dialog formspecs are much better now.

**Comment by ClobberXD** [comment]
As vertical real-estate is one of the major flaws in the current layout, increasing the height of the content store formspec would allow for more possibilities and/or further improved readability.

**Comment by paramat** [comment]
Drop-down moved to left edge or left where it is are both fine for me.

**Comment by random-geek** [comment]
I had initially placed the drop-down menu to the left of the scrollbar, but for some reason it was drawn underneath the images when expanded (even after putting it at the end of the code).

**Comment by paramat** [comment]
Ok, it's fine where it is then.

**Comment by rubenwardy** [comment]
how about putting it between 'back' and the page nav buttons? If you move back to the left

**Comment by random-geek** [comment]
The original post has been updated with new screenshots.

**Comment by paramat** [comment]
Looks good now thanks. Will review.

**Comment by paramat** [comment]
Still WIP?

**Comment by paramat** [comment]
Tested and code seems good.
One issue, when i select one of the options from the dropdown, it filters correctly but continues to display: 'all packages'.

File: builtin/mainmenu/dlg_contentstore.lua [hunk 1]
```diff
@@ -351,13 +357,14 @@ function store.get_formspec()
...
+		"field[0.2,0.1;7.8,1;search_string;;", core.formspec_escape(search_string), "]",
 		"field_close_on_enter[search_string;false]",
-		"button[10.2,-0.2;2,1;search;", fgettext("Search"), "]",
-		"dropdown[0,1;2.4;type;",
+		"dropdown[7.7,-0.1;2.4;type;",
 		table.concat(filter_types_titles, ","),
-		";",
+		";]",
+		"button[10.1,-0.2;2,1;search;", fgettext("Search"), "]",
 		filter_type,
```
Comment (paramat) [inline builtin/mainmenu/dlg_contentstore.lua]: In wrong place? Try moving to just after
```
table.concat(filter_types_titles, ","),
";",
```

Comment (paramat) [reply builtin/mainmenu/dlg_contentstore.lua]: Yes seems to work:
```
	local formspec = {
		"size[12,7;true]",
		"position[0.5,0.55]",
		"field[0.2,0.1;7.8,1;search_string;;", core.formspec_escape(search_string), "]",
		"field_close_on_enter[search_string;false]",
		"dropdown[7.7,-0.1;2.4;type;",
		table.concat(filter_types_titles, ","),
		";",
		filter_type,
		"]",
		"button[10.1,-0.2;2,1;search;", fgettext("Search"), "]",
		-- "textlist[0,1;2.4,5.6;a;",
		-- table.concat(taglist, ","),
		-- "]"
	}
```

**Comment by paramat** [comment]
I'm not keen on the position of the dropdown, this makes it look like it is a filter for search only, when it actually works independent of search.
I prefer on the left of the search bar, or at the bottom.

File: builtin/mainmenu/dlg_contentstore.lua (hunk 1 above)
Comment (random-geek) [reply builtin/mainmenu/dlg_contentstore.lua]: Not sure how that happened, will fix.

**Comment by paramat** [comment]
screenshot from 2018-11-07 00-59-50

Works now :+1:

**Comment by paramat** [comment]
@rubenwardy maybe you could provide the 2nd +1?

**Comment by paramat** [comment]
I think the dropdown should be at the top instead of the bottom, to avoid going off the screen bottom. So it seems to be in the optimum position.

**Comment by paramat** [comment]
Merge agreed by sfan5 http://irc.minetest.net/minetest-dev/2018-11-08#i_5435882

**Pull request merged**